howtos/cmake_install.rst

```diff
@@ -5,33 +5,69 @@ How to reuse cmake install for package() method
+``no_copy_source = True`` is set, affects the usage of conan with CMake.
+
+Conan retrieves the sources and put them in a source folder. Before compiling, conan creates a copy of the source folder
+into the build folder and starts the build. Once the build is done, conan creates the package by collecting the
+artifacts from the build folder using ``package()`` method. When ``no_copy_source = False``, there is nothing unexpected
+in the workflow.
+
+If you set ``no_copy_source = True``, conan still retrieves the sources, but does not copy them to the
+build folder. The build starts and once the artefacts are built, ``package()`` method needs to be called *twice*.
+First, source files are copied from the source folder into the package folder (which involves calling ``package()``
+once) and second, artifacts are copied from the build folder into the package folder (which again calls ``package()``
+for the second time).
+
+Please mind that CMake usually uses ``install`` directive to package both the artifacts and source code (*i.e.* header
+files) into the package folder. Hence calling ``package()`` twice, while having no side effects, is wasting a couple of
+cycles, since source code is already copied in the first invokation of ``package()`` and the install step will be done
+twice. Files will be simply overwritten, but install steps are sometimes time-expensive and this doubles the "packaging"
+time.
+
+This might be unintuitive if you only use CMake, but mind that conan needs to cater to many different build systems and
+scenarios (*e.g.* where you don't control the CMake configuration directly) and hence this workflow, though potentially
+unintuitive, is indispensable.
+
+Example
+-------
+The following excerpt shows how to build and package with CMake within conan. Mind that you need to configure CMake
+both in ``build()`` and in ``package()`` since these methods are called independently.
 
 .. code-block:: python
 
-    def build(self):
+    def configure_cmake(self):
         cmake = CMake(self)
         cmake.configure()
+        cmake.definitions["SOME_DEFINITION"] = "On"
+
+        return cmake
+
+    def build(self):
+        cmake = self.configure_cmake()
         cmake.build()
-        cmake.install()
-        # equivalent to
-        # args += ['-DCMAKE_INSTALL_PREFIX="%s"' % self.package_folder]
-        # self.run('cmake "%s/src" %s %s'
-        #          % (self.source_folder, cmake.command_line, ' '.join( args ) ) )
-        # self.run("cmake --build . --target install %s" % cmake.build_config)
 
     def package(self):
-        # nothing to do here now
+        cmake = self.configure_cmake()
+        cmake.install()
 
-The ``package_info()`` method is still necessary, as there is no possible way to
-automatically extract the information of the necessary libraries, defines and flags for different
-build configurations from the cmake install.
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
```

Comment:
These are the defaults, it is not necessary to define them, and it might be confusing to do so, without a clear warning or something.